### DISCUSSION TITLE
`remotePatterns` should accept `URL`s

### DISCUSSION THREAD

**Discussion by jeysal** [vercel/next.js#46241] [Ideas]
### Describe the feature you'd like to request

The `images.remotePatterns` config option currently accepts an object with the properties
* `protocol`
* `hostname`
* `port`
* `pathname`
which `URL` objects have.
However, passing a `URL` object is not possible because:
* It complains about excess properties such as `toString` and `href`
* It does not allow trailing `:` on the `protocol` property

### Describe the solution you'd like

It would be a nice quality-of-life improvement to be able to do something like `remotePatterns: [new URL(process.env.MY_IMAGE_HOST)]` directly.
To allow for this, Next should
* Deal with trailing `:` on the `protocol` property
* Not complain about excess properties if the remote pattern is `instanceof URL`

Both of these change are not breaking.

### Describe alternatives you've considered

Users might keep using the old `domains` property, passing `url.hostname`, because it is too much of a hassle to generate a `remotePatterns` object.

**Comment by balazsorban44** [comment]
Keep in mind that `remotePatterns` does not expect an exact URL, but a URL pattern, so [`URLPattern`](https://developer.mozilla.org/en-US/docs/Web/API/URLPattern) might be more correct, but it's missing from Node.js https://github.com/nodejs/node/issues/40844

Docs: https://nextjs.org/docs/api-reference/next/image#remote-patterns

**Reply by jeysal** [reply]
Yes, that would be nice to have on top (if they are 100% compatible). But even URL on its own would be a nice first step and cover many cases
